### PR TITLE
Helpers: Prevent silentScroll if user has already scrolled

### LINKED ISSUE #3958: Prevent silentScroll if user has already scrolled
Discussion about implementation details here:

https://github.com/jquery/jquery-mobile/pull/2095

### PR DESCRIPTION
**Pull request by apsdehal**
Fixes #3958
Closes #8448

### CODE REVIEW

File: js/helpers.js [hunk 1]
```diff
@@ -90,6 +90,12 @@ $.extend( $.mobile, {
 
 	// Scroll page vertically: scroll to 0 to hide iOS address bar, or pass a Y value
 	silentScroll: function( ypos ) {
+
+		// If user has already scrolled then do nothing
+		if ( $( window ) > 0 ) {
```
Comment (arschmitz) [inline js/helpers.js]: this will always be false since `$( window )` is not a number

**Comment by arschmitz** [comment]
Id like to see a test added to this

File: js/helpers.js (hunk 1 above)
Comment (apsdehal) [inline js/helpers.js]: Oops, this was to be `$( window ).scrollTop()`. Will fix.

File: tests/unit/helpers/helpers_core.js [hunk 2]
```diff
@@ -762,4 +762,39 @@ for ( index = 0; index < 4096; index++ ) {
...
+
+		$( window ).scrollTop( 150 );
+		assert.equal( $( window ).scrollTop(), 150, "At start, scroll is reset to 150" );
+		$.mobile.silentScroll( 0 );
+
+		window.setTimeout( function() {
+			assert.equal( $( window ).scrollTop(), 150,
+			"It should not scroll if user has already scrolled");
+			done();
+		}, 2000);
```
Comment (arschmitz) [inline tests/unit/helpers/helpers_core.js]: missing space

File: tests/unit/helpers/helpers_core.js [hunk 3]
```diff
@@ -762,4 +762,39 @@ for ( index = 0; index < 4096; index++ ) {
...
+
+	$.mobile.silentScroll( 100 );
+} );
+
+QUnit.test( "Silent scroll doesn't occur when scrollTop is not zero",
+	function ( assert ) {
+		assert.expect( 2 );
+		var done = assert.async();
+
+		$( window ).scrollTop( 150 );
```
Comment (arschmitz) [inline tests/unit/helpers/helpers_core.js]: $.mobile.window here and below

File: js/helpers.js [hunk 4]
```diff
@@ -90,6 +90,12 @@ $.extend( $.mobile, {
 
 	// Scroll page vertically: scroll to 0 to hide iOS address bar, or pass a Y value
 	silentScroll: function( ypos ) {
+
+		// If user has already scrolled then do nothing
+		if ( $( window ).scrollTop() > 0 ) {
```
Comment (arschmitz) [inline js/helpers.js]: $.mobile.window

File: tests/unit/helpers/helpers_core.js [hunk 5]
```diff
@@ -762,4 +762,39 @@ for ( index = 0; index < 4096; index++ ) {
 	createTestScenario( index );
 }
 
+QUnit.module( "Silent scroll tests" );
+
+QUnit.test( "Silent scroll works when scrollTop is zero", function( assert ) {
+	assert.expect( 3 );
+	var done = assert.async();
+
+	$( window ).scrollTop( 0 );
```
Comment (arschmitz) [inline tests/unit/helpers/helpers_core.js]: `$.mobile.window`

File: tests/unit/helpers/helpers_core.js [hunk 6]
```diff
@@ -762,4 +762,39 @@ for ( index = 0; index < 4096; index++ ) {
...
+		assert.expect( 2 );
+		var done = assert.async();
+
+		$( window ).scrollTop( 150 );
+		assert.equal( $( window ).scrollTop(), 150, "At start, scroll is reset to 150" );
+		$.mobile.silentScroll( 0 );
+
+		window.setTimeout( function() {
+			assert.equal( $( window ).scrollTop(), 150,
+			"It should not scroll if user has already scrolled");
```
Comment (arschmitz) [inline tests/unit/helpers/helpers_core.js]: spacing

**Comment by apsdehal** [comment]
@arschmitz Fixed.

File: js/helpers.js (hunk 4 above)
Comment (arschmitz) [inline js/helpers.js]: missed this one

**Comment by apsdehal** [comment]
I hope now its good

**Comment by arschmitz** [comment]
👍

**Pull request closed without merging**